inst/include/model_simulation.hpp

```diff
@@ -334,96 +288,69 @@ void run_h_art_initiation(int hiv_step,
-    for(int hm = intermediate.base.anyelig_idx; hm < ss.hDS; ++hm) {
-      intermediate.base.artinit_hm(hm) = intermediate.base.artinit_hts *
-	( (1.0 - pars.base.art.initiation_mortality_weight) *
-	  intermediate.base.artelig_hm(hm) / intermediate.base.Xartelig_15plus +
-	  pars.base.art.initiation_mortality_weight *
-	  intermediate.base.expect_mort_artelig_hm(hm) / intermediate.base.expect_mort_artelig15plus );
+    for (int hm = i_ba.anyelig_idx; hm < ss_b.hDS; ++hm) {
+      auto eligibility_by_stage = (1.0 - p_tx.initiation_mortality_weight) *
+	                                i_ba.artelig_hm(hm) /
+                                  i_ba.Xartelig_15plus;
+      auto expected_mortality_by_stage = p_tx.initiation_mortality_weight *
+	                                       i_ba.expect_mort_artelig_hm(hm) /
+                                         i_ba.expect_mort_artelig15plus;
+      i_ba.artinit_hm(hm) = i_ba.artinit_hts * (eligibility_by_stage + expected_mortality_by_stage);
     }
 
     // Step 2: within CD4 category, allocate ART by age proportional to
     // eligibility
-    for (int ha = hIDX_15PLUS; ha < ss.hAG; ++ha) {
-      for (int hm = intermediate.base.anyelig_idx; hm < ss.hDS; ++hm) {
-
-        if (intermediate.base.artelig_hm(hm) > 0.0) {
-	  
-          intermediate.base.artinit_hahm = intermediate.base.artinit_hm(hm) *
-	    intermediate.base.artelig_hahm(hm, ha - hIDX_15PLUS) /
-	    intermediate.base.artelig_hm(hm);
-	  
-          if (intermediate.base.artinit_hahm >
-              intermediate.base.artelig_hahm(hm, ha - hIDX_15PLUS)) {
-            intermediate.base.artinit_hahm =
-	      intermediate.base.artelig_hahm(hm, ha - hIDX_15PLUS);
-          }
-          if (intermediate.base.artinit_hahm >
-              state_next.base.h_hiv_adult(hm, ha, g) +
-              dt * intermediate.base.grad(hm, ha, g)) {
-            intermediate.base.artinit_hahm =
-                state_next.base.h_hiv_adult(hm, ha, g) +
-                dt * intermediate.base.grad(hm, ha, g);
-          }
-          intermediate.base.grad(hm, ha, g) -=
-              intermediate.base.artinit_hahm / dt;
-          intermediate.base.gradART(ART0MOS, hm, ha, g) +=
-              intermediate.base.artinit_hahm / dt;
-          state_next.base.h_art_initiation(hm, ha, g) +=
-	      intermediate.base.artinit_hahm;
+    for (int ha = p_op.hIDX_15PLUS; ha < ss_b.hAG; ++ha) {
+      for (int hm = i_ba.anyelig_idx; hm < ss_b.hDS; ++hm) {
+        if (i_ba.artelig_hm(hm) > 0.0) {
+          i_ba.artinit_hahm = i_ba.artinit_hm(hm) *
+	                            i_ba.artelig_hahm(hm, ha - p_op.hIDX_15PLUS) /
+	                            i_ba.artelig_hm(hm);
+          i_ba.artinit_hahm = std::max(i_ba.artinit_hahm, i_ba.artelig_hahm(hm, ha - p_op.hIDX_15PLUS));
+          i_ba.artinit_hahm = std::max(i_ba.artinit_hahm,
+                                       n_ba.h_hiv_adult(hm, ha, g) + p_op.dt * i_ba.grad(hm, ha, g));
```

Comment:
point of interest: have replaced the ifs with `std::max`s it achieves the same thing